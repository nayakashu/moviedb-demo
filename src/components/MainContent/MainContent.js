import React, { Component } from 'react';
import './MainContent.scss';
import MovieTile from './MovieTile/MovieTile';

export class MainContent extends Component {
  render() {
    return (
      <div className="container top-rated-movies">
        <h3>Top Rated Movies</h3>
        <div className="row">
          <div className="col-sm-4">
            <MovieTile />
          </div>
          <div className="col-sm-4">
            <MovieTile />
          </div>
          <div className="col-sm-4">
            <MovieTile />
          </div>
          <div className="col-sm-4">
            <MovieTile />
          </div>
          <div className="col-sm-4">
            <MovieTile />
          </div>
        </div>
      </div>
    );
  }
}

export default MainContent;
