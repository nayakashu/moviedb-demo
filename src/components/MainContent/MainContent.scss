@import '../../Common.scss';

.login-tran-enter {
  opacity: 0.01;
}

.login-tran-enter.login-tran-enter-active {
  opacity: 1;
  transition: opacity 300ms ease-in;
}

.login-tran-leave {
  opacity: 1;
}

.login-tran-leave.login-tran-leave-active {
  opacity: 0.01;
  transition: opacity 300ms ease-in;
}
