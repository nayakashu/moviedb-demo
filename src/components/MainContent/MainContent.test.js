import React from 'react';
import { shallow } from 'enzyme';
import MainContent from './MainContent';

describe('MainContent', () => {
  const mainContent = shallow(<MainContent />);

  it('renders correctly', () => {
    expect(mainContent).toMatchSnapshot();
  });
});
