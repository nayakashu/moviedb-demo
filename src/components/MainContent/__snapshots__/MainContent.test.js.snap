// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Top Rated Movies
        </h3>,
        <div
          className="row"
        >
          <div
            className="col-sm-4"
          >
            <MovieTile />
          </div>
          <div
            className="col-sm-4"
          >
            <MovieTile />
          </div>
          <div
            className="col-sm-4"
          >
            <MovieTile />
          </div>
          <div
            className="col-sm-4"
          >
            <MovieTile />
          </div>
          <div
            className="col-sm-4"
          >
            <MovieTile />
          </div>
        </div>,
      ],
      "className": "container top-rated-movies",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Top Rated Movies",
        },
        "ref": null,
        "rendered": "Top Rated Movies",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>,
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>,
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>,
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>,
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MovieTile />,
              "className": "col-sm-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MovieTile />,
              "className": "col-sm-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MovieTile />,
              "className": "col-sm-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MovieTile />,
              "className": "col-sm-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MovieTile />,
              "className": "col-sm-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Top Rated Movies
          </h3>,
          <div
            className="row"
          >
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>
            <div
              className="col-sm-4"
            >
              <MovieTile />
            </div>
          </div>,
        ],
        "className": "container top-rated-movies",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Top Rated Movies",
          },
          "ref": null,
          "rendered": "Top Rated Movies",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-sm-4"
              >
                <MovieTile />
              </div>,
              <div
                className="col-sm-4"
              >
                <MovieTile />
              </div>,
              <div
                className="col-sm-4"
              >
                <MovieTile />
              </div>,
              <div
                className="col-sm-4"
              >
                <MovieTile />
              </div>,
              <div
                className="col-sm-4"
              >
                <MovieTile />
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MovieTile />,
                "className": "col-sm-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MovieTile />,
                "className": "col-sm-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MovieTile />,
                "className": "col-sm-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MovieTile />,
                "className": "col-sm-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MovieTile />,
                "className": "col-sm-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
